### ISSUE TITLE
wifi-heatmap --ap-names seems not working

### ISSUE THREAD

**Issue by ggruening** [jantman/python-wifi-survey-heatmap#22]
Hello everyone,

Thanks for all the work!

I'm trying to use the `-a / --ap-names` option. Unfortunately I can't find any documentation about the structure of the json file. I tried something like this:

```
{
    "aa:bb:cc:dd:ee:01": "AP1 5",  
    "aa:bb:cc:dd:ee:02": "AP1 24",  
    "aa:bb:cc:dd:ee:03": "AP2 5",   
    "aa:bb:cc:dd:ee:04": "AP2 24",  
    "aa:bb:cc:dd:ee:05": "AP3 24", 
    "aa:bb:cc:dd:ee:06": "AP3 5"     
}
```

A `wifi-heatmap --ap-names ap-mapping.json [TITLE]` does not change the images at all. Is this a bug or am I doing something wrong?

Best regards,
Gregor

**Comment by jantman** [comment]
Gregor,

I don't actually use that option myself. I just looked through the code and it looks like, while this is not documented, ``--ap-names`` only works if ``wifi-heatmap`` is run with ``--show-points``. Could you please give that a try and let me know if that causes the AP names to show up?

Thanks,
Jason

**Comment by ggruening** [comment]
Thanks for the hint.

The measurement points are displayed with `--show-points`, but their labeling is in the form `<SSID> (<frequency>)`. The name of the access points is still missing.

There is actually no difference between `wifi-heatmap --show-points [TITLE]` and `wifi-heatmap --show-points --ap-names ap-mapping.json [TITLE]`.

Are there any further ideas?

**Comment by jantman** [comment]
Hmmm... I'll need to take a look after work today. I don't use the ap names feature myself, so it could just be broken in a way that I haven't noticed yet.

Is there any chance that you could upload one of the resulting heatmaps, and also the output of running that command with `-vv` for debug-level output?

**Comment by ggruening** [comment]
You're welcome :-)

[heatmap.log](https://github.com/jantman/python-wifi-survey-heatmap/files/8373354/heatmap.log)


![tcp_download_Mbps_example json](https://user-images.githubusercontent.com/41573825/160661623-c51bdde0-afd2-4688-9473-42703aa4dc1d.png)

**Comment by jantman** [comment]
Looking at the code, I *think* the JSON file expects SSID keys, not BSSID.

In `wifi-heatmap`, the contents of `ap-mapping.json` is [loaded\(https://github.com/jantman/python-wifi-survey-heatmap/blob/master/wifi_survey_heatmap/heatmap.py#L151-L156) into `self._ap_names`, with the keys converted to upper-case. When [processing the survey data](https://github.com/jantman/python-wifi-survey-heatmap/blob/master/wifi_survey_heatmap/heatmap.py#L240-L244) (`survey_points` list in the JSON file, i.e. `example.json`), for each item in `survey_points`, we look at the ``ssid`` value in the ``result`` dictionary. If there's a matching key (upper case to upper case) in the AP names, we display that value. If not, we display the SSID.

**Comment by ggruening** [comment]
Maybe I'm confused, but I do not see the point. A network that is broadcast from several access points has the _same SSID_ everywhere, and the information that would be helpful is which access point I am actually connected to _despite_ the same SSID. What am I missing?

**Comment by jantman** [comment]
You're not confused. It appears that this is a bug.

Looking back in the git history, it [originally](https://github.com/jantman/python-wifi-survey-heatmap/commit/869b6f767fc93bd898f5ad1f0f7da243923a4629) used BSSIDs like you'd expect. The README even has a line about that:

> You can optionally pass the path to a JSON file mapping the access point MAC addresses (BSSIDs) to friendly names via the ``-a`` / ``--ap-names`` argument. If specified, this will annotate each measurement dot on the heatmap with the name (mapping value) and frequency band of the AP that was connected when the measurement was taken. This can be useful in multi-AP roaming environments.

However, v1.0.0 / #4 was a PR that included a massive rewrite of the wireless-handling code to move from iwlist to nl80211. It appears that [a commit in that PR](https://github.com/jantman/python-wifi-survey-heatmap/commit/7dab99dd523336e07a0e5d0c5ff0d78ae5ecac99) switched these labels from BSSID to SSID.

**Comment by jantman** [comment]
I have a fix for this currently in a branch: https://github.com/jantman/python-wifi-survey-heatmap/tree/issues/22

I'll try to get that released in the next few days.

**Comment by ggruening** [comment]
Just checked out the new branch... it works! Thank you very much. It makes this great software even more great :-) I'll stay tuned for the new release.

**Comment by jantman** [comment]
This has been released as 1.1.0 and should be live on Docker Hub as well. Thank you so much for taking the time to report the issue and help with troubleshooting!
